### ISSUE TITLE
When data input parameters are allowed to be null, allow null

### ISSUE THREAD

**Issue by dallenk** [Cacti/cacti#5585]
# Describe the bug

[A clear and concise description of what the bug is.]
https://github.com/Cacti/cacti/issues/4649 references this bug and it appears to have returned with the most recent devel pull.

## To Reproduce

Steps to reproduce the behavior with a fresh installation

1. Create a discovery network
2. set up criteria to add "Advanced Ping" 
3. discover the network and watch logs.
4. See error.

## Additional context
![image](https://github.com/Cacti/cacti/assets/2725066/764923db-1957-4221-b220-f12bd09ab724)

```
2023-11-18 19:56:04 - AUTOM8 NOTE: Device[MF8500C Series], GT[PING - Advanced Ping] Graph not added due to no default value for overridable field.
2023-11-18 19:56:04 - AUTOM8 NOTE: Device[[MF8500C Series](http://devel.cacti.domain.com/host.php?action=edit&id=27)], GT[[PING - Advanced Ping](http://devel.cacti.domain.com/graph_templates.php?action=template_edit&id=286)] Graph not added due to no default value for overridable field.
```

**Comment by dallenk** [comment]
this also seems to crash and disable the thold plugin when enabled.


```
2023-11-18 20:30:46 - CMDPHP ERRORS DETECTED - DISABLING PLUGIN 'thold'
2023-11-18 20:30:46 - PHP ERROR Backtrace: (CactiShutdownHandler())
2023-11-18 20:30:46 - ERROR PHP ERROR in Plugin 'thold': Uncaught Error: Call to undefined function autocreate() in /var/www/cacti_devel/plugins/thold/setup.php:1526 Stack trace: #0 /var/www/cacti_devel/lib/plugins.php(281): thold_device_autocreate() #1 /var/www/cacti_devel/lib/plugins.php(158): api_plugin_run_plugin_hook_function() #2 /var/www/cacti_devel/lib/api_automation.php(3463): api_plugin_hook_function() #3 /var/www/cacti_devel/poller_automation.php(766): automation_add_device() #4 /var/www/cacti_devel/poller_automation.php(407): discoverDevices() #5 {main} thrown in file: /var/www/cacti_devel/plugins/thold/setup.php on line: 1526
2023-11-18 20:30:45 - AUTOM8 NOTE: Device[MF8500C Series], GT[PING - Advanced Ping] Graph not added due to no default value for overridable field.
2023-11-18 20:30:45 - SQL Backtrace: (/poller_automation.php[407]:discoverDevices(), /poller_automation.php[759]:db_fetch_cell(), /lib/database.php[659]:db_fetch_cell_prepared(), /lib/database.php[686]:db_execute_prepared())
2023-11-18 20:30:45 - CMDPHP ERROR: A DB Cell Failed!, Error: Unknown column 'MF8500C' in 'field list'
2023-11-18 20:30:45 - PHP ERROR WARNING Backtrace: (/poller_automation.php[407]:discoverDevices(), /poller_automation.php[755]:CactiErrorHandler())
2023-11-18 20:30:45 - ERROR PHP WARNING: Undefined variable $dns_shortname in file: /var/www/cacti_devel/poller_automation.php on line: 755

2023-11-18 20:30:46 - CMDPHP ERRORS DETECTED - DISABLING PLUGIN 'thold'
2023-11-18 20:30:46 - PHP ERROR Backtrace: (CactiShutdownHandler())
2023-11-18 20:30:46 - ERROR PHP ERROR in Plugin 'thold': Uncaught Error: Call to undefined function autocreate() in /var/www/cacti_devel/plugins/thold/setup.php:1526 Stack trace: #0 /var/www/cacti_devel/lib/plugins.php(281): thold_device_autocreate() #1 /var/www/cacti_devel/lib/plugins.php(158): api_plugin_run_plugin_hook_function() #2 /var/www/cacti_devel/lib/api_automation.php(3463): api_plugin_hook_function() #3 /var/www/cacti_devel/poller_automation.php(766): automation_add_device() #4 /var/www/cacti_devel/poller_automation.php(407): discoverDevices() #5 {main} thrown in file: /var/www/cacti_devel/plugins/thold/setup.php on line: 1526
2023-11-18 20:30:45 - AUTOM8 NOTE: Device[[MF8500C Series](http://devel.cacti.pghost.com/host.php?action=edit&id=28)], GT[[PING - Advanced Ping](http://devel.cacti.pghost.com/graph_templates.php?action=template_edit&id=286)] Graph not added due to no default value for overridable field.
2023-11-18 20:30:45 - SQL Backtrace: (/poller_automation.php[407]:discoverDevices(), /poller_automation.php[759]:db_fetch_cell(), /lib/database.php[659]:db_fetch_cell_prepared(), /lib/database.php[686]:db_execute_prepared())
2023-11-18 20:30:45 - CMDPHP ERROR: A DB Cell Failed!, Error: Unknown column 'MF8500C' in 'field list'
2023-11-18 20:30:45 - PHP ERROR WARNING Backtrace: (/poller_automation.php[407]:discoverDevices(), /poller_automation.php[755]:CactiErrorHandler())
2023-11-18 20:30:45 - ERROR PHP WARNING: Undefined variable $dns_shortname in file: /var/www/cacti_devel/poller_automation.php on line: 755

```
another side effect is what appears to be a running process that won't end.  deleting the network and recreating is seems to work.
![image](https://github.com/Cacti/cacti/assets/2725066/b06ced48-a85e-4979-9670-7bc1500d4783)

**Comment by TheWitness** [comment]
I fixed this here for `develop` 5217f37999612757efb0b5c17b2a0fdf420fc8ba
